### DISCUSSION TITLE
Should formulas be transformed to uppercase?

### DISCUSSION THREAD

**Discussion by jansiegel** [handsontable/hyperformula#40] [General]
### Question:
If the user passes `=Sum(1;2)` to a cell, should it be transformed to `=SUM(1;2)`?

**Excel** *does* trasform it to `=SUM(1;2)`, **Google Spreadsheet** *doesn't*.

**Comment by wojciechczerniak** [comment]
@jansiegel You’re asking if you should do it on Handsontable side? No. Engine parser will handle this. 

When it comes to rules, they are defined in spec:

> Function names ignore case. Evaluators should write function names in all uppercase letters when writing OpenFormula formulas.

And another one:

> Evaluators should retain whitespace entered by the original formula creator and use it when saving or presenting the formula, and should not add additional whitespace unless directed to do so during the process of editing a formula.

But I have to ask @voodoo11  and @swistak35 how they handle it in the AST parser. 

Source:
http://docs.oasis-open.org/office/v1.2/os/OpenDocument-v1.2-os-part2.html#__RefHeading__1017860_715980110

**Comment by jansiegel** [comment]
What I meant is:
1. User enters `=Sum(A1,B2)` to the cell editor, it gets sent to hyperformula
2. User opens the editor again, HOT calls hyperformula's `getCellFormula` and receives `=SUM(A1,B2)`

**Comment by wojciechczerniak** [comment]
Yup, that's how it should work.

**Comment by wojciechczerniak** [comment]
Parser can handle lower case:

https://github.com/handsontable/hyperformula/blob/d7fffcebd683f685214a23b312c5cb76bcb7422a/test/parser/parser.spec.ts#L218-L223

Unparser will rewrite to upper case:

https://github.com/handsontable/hyperformula/blob/d7fffcebd683f685214a23b312c5cb76bcb7422a/test/parser/unparse.spec.ts#L133-L140

As stated in ODFF spec 👍
